src/gui/symbology-ng/qgsgraduatedsymbolrendererv2widget.cpp

```diff
@@ -446,6 +453,16 @@ QgsGraduatedSymbolRendererV2Widget::QgsGraduatedSymbolRendererV2Widget( QgsVecto
   btnAdvanced->setMenu( advMenu );
 }
 
+void QgsGraduatedSymbolRendererV2Widget::on_mSizeUnitWidget_changed()
+{
+  if ( !mGraduatedSymbol ) return;
+  mGraduatedSymbol->setOutputUnit( QgsSymbolV2::OutputUnit( mSizeUnitWidget->getUnit() ) );
```

Comment:
use QgsUnitSelectionWidget::unit() here instead. getUnit is horribly fragile as it depends on the order of units in the combo box matching the order of QgsSymbolV2::OutputUnit